File: app/src/main/java/org/stepic/droid/web/EmptyAuthService.java
package org.stepic.droid.web;

import org.stepic.droid.util.AppConstants;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;

public interface EmptyAuthService {

    @GET("/")
    Call<Void> getStepicForFun(@Header("Accept-Language") String lang);

    @FormUrlEncoded
    @POST("accounts/password/reset/")
    Call<Void> remindPassword(@Field(value = "email", encoded = true) String email);

    @GET("api/stepics/1")
    Call<StepicProfileResponse> getUserProfileWithCookie(@Header(AppConstants.refererHeaderName) String referer,
                                                         @Header(AppConstants.cookieHeaderName) String cookies,
                                                         @Header(AppConstants.csrfTokenHeaderName) String csrfToken);


}

File: app/src/main/java/org/stepik/android/remote/auth/service/EmptyAuthService.kt
package org.stepik.android.remote.auth.service

import org.stepic.droid.util.AppConstants
import org.stepik.android.remote.auth.model.StepikProfileResponse
import retrofit2.Call
import retrofit2.http.Field
import retrofit2.http.FormUrlEncoded
import retrofit2.http.GET
import retrofit2.http.Header
import retrofit2.http.POST

interface EmptyAuthService {

    @GET("/")
    fun getStepicForFun(@Header("Accept-Language") lang: String): Call<Void>

    @FormUrlEncoded
    @POST("accounts/password/reset/")
    fun remindPassword(@Field(value = "email", encoded = true) email: String): Call<Void>

    @GET("api/stepics/1")
    fun getUserProfileWithCookie(
        @Header(AppConstants.refererHeaderName) referer: String,
        @Header(AppConstants.cookieHeaderName) cookies: String,
        @Header(AppConstants.csrfTokenHeaderName) csrfToken: String
    ): Call<StepikProfileResponse>
}